crates/pathfinder/src/state/sync.rs

```diff
@@ -775,6 +822,18 @@ fn deploy_contract(
         .context("Inserting class hash into contracts table")
 }
 
+fn replace_class(
+    transaction: &Transaction<'_>,
+    replaced_class: &ReplacedClass,
+) -> anyhow::Result<()> {
+    ContractsTable::upsert(
+        transaction,
+        replaced_class.address,
+        replaced_class.class_hash,
+    )
+    .context("Updating class hash in contracts table")
```

Comment:
Is having a single entry per table good enough? Should we not be flattening the structure to include block changed at etc? Haven't seen how reorgs will be handled here, maybe that comes in a later change still.

Reply:
Hmm, that's a good point... We didn't need it because contracts were immutable once deployed, but with class replacements this won't properly work with reorgs.

I was wondering if we need this table at all... Looks like the only place we're using it is in `update_contract_state()` where we're fetching the class hash for the contract address so that we can re-calculate the contract state hash. But we could just do that based on `ContractsStateTable` from where we're fetching the old contract state root and nonce anyway?

Reply:
Yes I think you're correct that we don't actually need it.

I do think we will want to "flatten" this (along with the state storage updates) so that we no longer need all the trees. However, I suppose that can be a (massive) future migration. Or we just wait for regenesis.

Reply:
I've added some commits that remove this table. I think this also solves the issue with reorgs.